openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -75,48 +79,263 @@ export function closePop(){
+    books(work) {
+        const author_name = work.author_name ? work.author_name[0] : '';
+        $('header#header-bar .search-component ul.search-results').append(
+            `<li class="instant-result">
+                <a href="${work.key}">
+                    <img src="//covers.openlibrary.org/b/id/${work.cover_i}-S.jpg"/>
+                    <span class="book-desc">
+                        <div class="book-title">${work.title}</div> by <span class="book-author">${author_name}</span>
+                    </span>
+                </a>
+            </li>`
+        );
+    },
+    authors(author) {
+        // Todo: default author img to: https://dev.openlibrary.org/images/icons/avatar_author-lg.png
+        $('header#header-bar .search-component ul.search-results').append(
+            `<li>
+                <a href="/authors/${author.key}">
+                    <img src="http://covers.openlibrary.org/a/olid/${author.key}-S.jpg"/>
+                    <span class="author-desc"><div class="author-name">${author.name}</div></span>
+                </a>
+            </li>`
+        );
+    }
+}
 
-    $(window).scroll(function(){
-        var scroller = $('#formScroll');
-        if(isScrolledIntoView(scroller)){$('#scrollBtm').show();}else{$('#scrollBtm').hide();}
-    });
+/** Manages search state variables */
+class SearchState {
+    constructor(urlParams) {
+        this._listeners = {};
+
+        if (!(this.facet in SEARCH_FACETS)) {
+            this.facet = DEFAULT_SEARCH_FACET;
+        }
+        this.facet = urlParams.facet || this.facet || DEFAULT_SEARCH_FACET;
+        this.searchMode = urlParams.mode;
+    }
+
+    get facet() {
+        return localStorage.getItem('facet');
+    }
+    set facet(newFacet) {
+        const oldValue = this.facet;
+        localStorage.setItem('facet', newFacet);
+        this._trigger('facet', newFacet, oldValue);
+    }
+
+    get searchMode() {
+        return localStorage.getItem('mode');
+    }
+    set searchMode(mode) {
+        const oldValue = this.searchMode;
+        const searchMode = (mode && mode.toLowerCase()) || oldValue;
+        const isValidMode = SEARCH_MODES.indexOf(searchMode) != -1;
+        const newMode = isValidMode ? searchMode : SEARCH_MODE_DEFAULT;
+        localStorage.setItem('mode', newMode);
+        this._trigger('searchMode', newMode, oldValue);
+    }
+
+    sync(key, handler, user_opts={}) {
+        const DEFAULT_OPTS = {
+            fireAtStart: true,
+            onlyFireOnChange: true
+        };
+
+        if (!(key in this))
+            throw Error('Invalid key', key);
+
+        const opts = Object.assign({}, DEFAULT_OPTS, user_opts);
+        this._listeners[key] = this._listeners[key] || [];
+        this._listeners[key].push({ handle: handler, opts });
+        if (opts.fireAtStart) handler(this[key]);
+    }
+
+    /**
+     * @param {String} key
+     * @param {any} newValue
+     * @param {any} oldValue
```

Comment:
What does this do? Please explain the function. @param only get you so far.

Reply:
This gets axed as well :P